flang/include/flang/Optimizer/Transforms/Passes.td

```diff
@@ -43,6 +43,20 @@ def AffineDialectDemotion : FunctionPass<"demote-affine"> {
   let constructor = "fir::createAffineDemotionPass()";
 }
 
+def FirLoopResultOpt : FunctionPass<"fir-loop-result-opt"> {
+    let constructor = "fir::createFirLoopResultOptPass()";
```

Comment:
Maybe add a summary and description

Reply:
okay, also I am not sure about the name of this pass.

Reply:
Ah. Maybe something like "DCE loop results"? That may be worse though. heh